zerver/lib/user_groups.py

```diff
@@ -293,13 +293,13 @@ def get_direct_user_groups(user_profile: UserProfile) -> List[UserGroup]:
 def get_user_group_direct_member_ids(
     user_group: UserGroup,
 ) -> ValuesQuerySet[UserGroupMembership, int]:
-    return UserGroupMembership.objects.filter(user_group=user_group).values_list(
-        "user_profile_id", flat=True
-    )
+    return UserGroupMembership.objects.filter(
+        user_group=user_group, user_profile__is_active=True
+    ).values_list("user_profile_id", flat=True)
 
 
 def get_user_group_direct_members(user_group: UserGroup) -> QuerySet[UserProfile]:
-    return user_group.direct_members.all()
+    return user_group.direct_members.filter(is_active=True)
```

Comment:
This function is currently only callled in `is_user_in_group` and we can be sure that the user passed to it would be active, so filtering is not required here, but did this change for consistency and to make sure we do not encounter any bugs if this function is used at any other place in future.

Reply:
Yeah definitely the safe approach.